src/main/java/com/uid2/admin/audit/QLDBAuditWriter.java

```diff
@@ -37,20 +36,35 @@ public void writeLog(AuditModel model) {
         }
         try {
             if (qldbLogging) {
-                qldbDriver.execute(txn -> {
-                    List<IonValue> sanitizedInputs = new ArrayList<>();
-                    JsonObject jsonObject = model.writeToJson();
-                    StringBuilder query = new StringBuilder("UPDATE " + logTable + " AS t SET data = ?");
-                    sanitizedInputs.add(ionSys.newLoader().load(jsonObject.toString()).get(0));
-                    query.append(" WHERE t.data.itemType = ? AND t.data.itemKey = ?");
-                    sanitizedInputs.add(ionSys.newString(jsonObject.getString("itemType")));
-                    sanitizedInputs.add(ionSys.newString(jsonObject.getString("itemKey")));
+                JsonObject jsonObject = model.writeToJson();
+                if(jsonObject.getString("actionTaken").equals("CREATE")){
```

Comment:
Why do you write this to JSON then look up the field, couldn't you just check this field from the class?

Reply:
I either do this or assert that model instanceof QLDBAuditWriter. To be fair, this is probably a fair assertion as there's something wrong if a different audit model is being passed :)